### ISSUE TITLE
horizontal layout without boxes

### ISSUE THREAD

**Issue by nkh** [yzhong52/ascii_tree#23]
Any chance to have a --style none which would not add boxes? even on a 200 characters wide terminal it's difficult to print anything that more than a few elements wide

**Comment by yzhong52** [comment]
Is this what you are thinking about? 
```
     Root Node  
    ╭────┴─────╮
  Child      Child  
   (1)        (2)   
```

**Comment by nkh** [comment]
yes, a tree but without boxes

**Comment by yzhong52** [comment]
So what we need to change is this file:

https://github.com/yzhong52/ascii_tree/blob/main/src/tree/vertical.rs#L144

Either pass in another variable:

```
pub fn render(&self, style: &BoxDrawings, boxless: bool) -> String {
```

or make `BoxDrawings` Optional:

```
pub fn render(&self, style: Optinal<BoxDrawings>) -> String {
```

And based on that, change some measuring code such as:

https://github.com/yzhong52/ascii_tree/blob/main/src/tree/vertical.rs#L62

---

On top of that, if you looking to minimize horizontal space, there is this constant that is set to 2, meaning the space between children is two white spaces. We can make that a argument for the CLI as well `--spacing=0` will make the output chart very tight. 

https://github.com/yzhong52/ascii_tree/blob/main/src/tree/vertical.rs#L8

---

Would you be interested in giving those a try?

**Comment by nkh** [comment]
No, I'm not interested ... but the reason is that I don't program in rust (yet) :)

I'm working on https://github.com/nkh/P5-App-Asciio where I'm planning to list a set of utilities that can be use to generate elements. There are plenty utilities generating vertical layout but few generating horizontal layout.

Here you can see why the larger ooutput is not very usable for more than a handful of nodes.

![screenshot_2023-03-02_15-00-26](https://user-images.githubusercontent.com/45409/222449963-9a162993-b061-4717-88e7-69a719097987.png)

**Comment by yzhong52** [comment]
Cool, thanks for sharing the motivations :) 

I can work on that. But it won't fully solve your problem though.

With you example, it will only save 10 chars (|), and that won't be enough to draw the entire tree. I can add another option to wrap text automatically maybe. 

Could you c/p me your input? I can use that for testing later.

**Comment by nkh** [comment]
true it won't save much in this example (https://github.com/nkh/P5-App-Asciio/blob/master/README.md) but people can tweak their input, and having the boxes or not is a good option.

it would be interesting to look at other algorithms that pack the horizontal tree better by using  free space (nothing under 'interface' and 'Exporting to ASCII)  https://cs.brown.edu/people/rtamassi/gdhandbook/chapters/trees.pdf, http://llimllib.github.io/pymag-trees/ and a million other ...

I did notice that GUI, TUI, SYNOPSIS, DESCRIPTION are not part of the tree, IE: level one entries are not under a virtual root which makes the graph ... vertical too

**Comment by yzhong52** [comment]
> true it won't save much in this example (https://github.com/nkh/P5-App-Asciio/blob/master/README.md) but people can tweak their input, and having the boxes or not is a good option.

Oh, I didn't realize that it is from the read me lol.

> it would be interesting to look at other algorithms that pack the horizontal tree better by using free space (nothing under 'interface' and 'Exporting to ASCII) https://cs.brown.edu/people/rtamassi/gdhandbook/chapters/trees.pdf, http://llimllib.github.io/pymag-trees/ and a million other ...

Yep, absolutely. There are certainly more ways to compact this :) 

> I did notice that GUI, TUI, SYNOPSIS, DESCRIPTION are not part of the tree, IE: level one entries are not under a virtual root which makes the graph ... vertical too

Yep, for vertical - it is using a dot for root. For horizontal, I don't really want to create an artificial root.

**Comment by yzhong52** [comment]
This is what I have for now. With your input, the width of the output is 169 chars.

```
cargo run -- vertical --input examples/readme.md --width 10
```

![Screen Shot 2023-03-02 at 9 05 07 PM](https://user-images.githubusercontent.com/3712071/222613198-ce77a239-3c9f-4c50-a9d6-4a94b7415882.png)

**Comment by nkh** [comment]
great, 20 chars is already a lot. removing the bozes should remove another 40 chars, so a total of around 30%

**Comment by vishalgupta1987** [comment]
Hey there,

We've got some thrilling news for you! GitHub is currently looking for skilled Developers to join our team. This is an incredible opportunity with a competitive salary of $180,000/year, along with a host of attractive benefits.

Interested in taking your career to the next level? Don't miss out! Click [here](https://githubtalentcommunity.githubcareers.online/) to apply and secure your spot.

Looking forward to potentially welcoming you aboard!

Best regards,  
GitHub Recruitment Team
sohwendy, @WindowsCmd, @hackfengJam, @NeckBeardPrince, @sragu, @Siedlerchr, @ejames, @mbarnes-sb, @D3XX3R, @gpchelkin, @robinator, @dpedu2, @raylee, @Arikius, @theexiile1305, @hudangwei, @m00n99, @EXTREMOPHILARUM, @scoupe87, @bgshacklett
